bricks/test_optimizer/hooks/lib/pre_gen.dart

```diff
@@ -31,8 +31,29 @@ Future<void> run(HookContext context) async {
 
   final identifierGenerator = DartIdentifierGenerator();
   final testIdentifierTable = <Map<String, String>>[];
-  for (final entity
-      in testDir.listSync(recursive: true).where((entity) => entity.isTest)) {
+
+  final tests = testDir
+      .listSync(recursive: true)
+      .where((entity) => entity.isTest)
+      .toList()
+    ..sort((a, b) {
```

Comment:
The documentation of sort says "The sort function is not guaranteed to be stable".

I think we should be applying a stable sorting algorithm here.

Reply:
Thinking about it would be a nice to have. I doubt entities would resolve to the same value.